{{ cookiecutter.format }}/{{ cookiecutter.formal_name }}.app/Contents/Info.plist

```diff
@@ -29,56 +29,75 @@
 	{% if cookiecutter.document_types %}
 	<key>CFBundleDocumentTypes</key>
 	<array>
-		{% for extension, doctype in cookiecutter.document_types.items() %}
+		{% for document_type_id, document_type in cookiecutter.document_types.items() %}
+		{% set macOS = document_type.macOS | default({}) %}
 		<dict>
-			<key>CFBundleTypeName</key>
-			<string>{{ doctype.description }}</string>
-			<key>CFBundleTypeIconFile</key>
-			<string>{{ cookiecutter.app_name }}-{{ extension }}.icns</string>
-			<key>CFBundleTypeExtensions</key>
-			<array>
-				<string>{{ extension }}</string>
-			</array>
-			<key>CFBundleTypeMIMETypes</key>
-			<array/>
 			<key>CFBundleTypeRole</key>
-			<string>Viewer</string>
+			<string>{{ macOS.CFBundleTypeRole | default("Editor") }}</string>
+			<key>LSHandlerRank</key>
+			<string>{{ macOS.LSHandlerRank | default("Owner") }}</string>
 			<key>LSItemContentTypes</key>
 			<array>
-				<string>{{ cookiecutter.bundle }}.{{ cookiecutter.app_name }}.{{ extension }}</string>
+				{%- if macOS.LSItemContentTypes is defined -%}
+				{% for content_type in macOS.LSItemContentTypes %}
+				<string>{{ content_type }}</string>
+				{%- endfor %}
+				{%- else %}
+				<string>{{ cookiecutter.bundle }}.{{ cookiecutter.app_name }}.{{ document_type_id }}</string>
+				{%- endif %}
 			</array>
 			<key>LSTypeIsPackage</key>
-			<integer>1</integer>
+			{{ "<true/>" if macOS.LSTypeIsPackage | default(False) else "<false/>"}}
```

Comment:
The `<true/>`/`<false/>` rendering can come out of the `|plist_value` filter provided by Briefcase.

Reply:
Will fix this tomorrow!